substrate/primitives/consensus/sassafras/Cargo.toml

```diff
@@ -0,0 +1,64 @@
+[package]
+name = "sp-consensus-sassafras"
+version = "0.3.4-dev"
+authors = ["Parity Technologies <admin@parity.io>"]
+description = "Primitives for Sassafras consensus"
+edition = "2021"
+license = "Apache-2.0"
+homepage = "https://substrate.io"
+repository = "https://github.com/paritytech/substrate/"
+documentation = "https://docs.rs/sp-consensus-sassafras"
+readme = "README.md"
+publish = false
+
+[package.metadata.docs.rs]
+targets = ["x86_64-unknown-linux-gnu"]
+
+[dependencies]
+scale-codec = { package = "parity-scale-codec", version = "3.2.2", default-features = false }
+scale-info = { version = "2.5.0", default-features = false, features = ["derive"] }
+serde = { version = "1.0.163", default-features = false, features = ["derive"], optional = true }
+sp-api = { version = "4.0.0-dev", default-features = false, path = "../../api" }
+sp-application-crypto = { version = "23.0.0", default-features = false, path = "../../application-crypto", features = ["bandersnatch-experimental"] }
+sp-consensus-slots = { version = "0.10.0-dev", default-features = false, path = "../slots" }
+sp-core = { version = "21.0.0", default-features = false, path = "../../core", features = ["bandersnatch-experimental"] }
+sp-runtime = { version = "24.0.0", default-features = false, path = "../../runtime" }
+sp-std = { version = "8.0.0", default-features = false, path = "../../std" }
+
+# Temporary workaround, remove as soon as 'bandersnatch' is stable.
+#
+# Sassafras requires to enable `bandersnatch-experimental` feature for some of its dependencies
+# and this ends up being transitively enabled in `sp-keystore` crate.
+#
+# E.g. sp-application-crypto → sp-io → sp-keystore
+#
+# Follows that Bandersnatch api ends up being exposed by the `Keystore` trait and thus we require
+# to expose these functions implementations in the client keystore (`sc-keystore`) as well.
+#
+# The crate feature should be enabled only for `std` builds (keep it disabled by default).
+sc-keystore = { path = "../../../client/keystore", optional = true}
```

Comment:
https://github.com/paritytech/polkadot-sdk/blob/7d77005522e4138eef9a7b55850ef4a5aa72d8ff/substrate/primitives/core/src/lib.rs#L404-L424 use this macro to create a `bandersnatch_enabled` macro to guard the implementation of the trait methods in `sc-keystore`.

Reply:
I wasn't aware of this. Is doing the job better. Thank you